malfeasance/handler.go

```diff
@@ -31,37 +29,51 @@ var (
 	errInvalidProof  = fmt.Errorf("%w: invalid proof", pubsub.ErrValidationReject)
 )
 
+type (
+	MalfeasanceHandlerV1 func(ctx context.Context, data scale.Type) (types.NodeID, []string, error)
+	MalfeasanceHandlerV2 func(ctx context.Context, data []byte) (types.NodeID, []string, error)
+)
+
 // Handler processes MalfeasanceProof from gossip and, if deems it valid, propagates it to peers.
 type Handler struct {
-	logger       log.Log
-	cdb          *datastore.CachedDB
-	self         p2p.Peer
-	nodeIDs      []types.NodeID
-	edVerifier   SigVerifier
-	tortoise     tortoise
-	postVerifier postVerifier
+	logger *zap.Logger
+	cdb    *datastore.CachedDB
+
+	handlersV1 map[uint8]MalfeasanceHandlerV1
```

Comment:
How about defining a 
```go
type Type uint8
```
and
```suggestion
	handlersV1 map[Type]MalfeasanceHandlerV1
```

and also have it in:
```go

func (h *Handler) RegisterHandlerV1(malfeasanceType Type, handler MalfeasanceHandlerV1) {
```

Reply:
I was thinking about that as well. My last commit contains an update that makes it a dedicated type and decouples it from the encoding (which it always should have been) but preserves backwards compatibility.